### PR TITLE
Remove rest client

### PR DESCRIPTION
**Pull request by Kallin**
rest-client is no longer maintained
use faraday instead

### CODE REVIEW

**Comment by Kallin** [comment]
thx @dubroe , yeah I saw that and removed the review request 🙈

**Comment by Kallin** [comment]
@dubroe should be good now, sorry about that!

File: Gemfile.lock [hunk 1]
```diff
@@ -2,43 +2,42 @@ PATH
   remote: .
   specs:
     marqeta (0.1.1)
-      rest-client
+      faraday
 
 GEM
   remote: https://rubygems.org/
   specs:
-    ast (2.4.0)
```
Comment (dubroe) [inline Gemfile.lock:10]: Did you expect all of this to change just by replacing this one gem?

File: Rakefile [hunk 2]
```diff
@@ -1,3 +1,5 @@
+# frozen_string_literal: true
```
Comment (dubroe) [inline Rakefile:1]: Maybe we can put these changes in a separate PR since it's unrelated?

File: .rubocop_todo.yml [hunk 3]
```diff
@@ -1,48 +1,42 @@
...
+# This cop supports safe autocorrection (--autocorrect).
 Layout/EmptyLineAfterGuardClause:
   Exclude:
     - 'lib/marqeta/user.rb'
 
 # Offense count: 1
-# Cop supports --auto-correct.
-Layout/LeadingBlankLines:
+# Configuration parameters: AllowedParentClasses.
+Lint/MissingSuper:
```
Comment (dubroe) [inline .rubocop_todo.yml:24]: Can you please explain why this Rubocop file is being changed here?

File: lib/marqeta/api_object.rb [hunk 4]
```diff
@@ -32,7 +35,7 @@ def self.object_list(klass, endpoint)
       while is_more
         paginated_params = {
           count: QUERY_RESULTS_COUNT,
-          start_index: start_index
+          start_index:
```
Comment (dubroe) [inline lib/marqeta/api_object.rb:38]: Can you have a hash key without a value like this?

**Comment by dubroe** [comment]
@Kallin I think it would be cleaner to try to make this PR be just about replacing rest client with Faraday and then making a separate PR for these rubocop changes.

File: lib/marqeta/api_object.rb (hunk 4 above)
Comment (dubroe) [reply lib/marqeta/api_object.rb:38]: Ah cool I hadn't seen this https://blog.saeloun.com/2021/09/28/ruby-allow-value-omission-in-hash-literals/

**Comment by Kallin** [comment]
ok sure @dubroe I'll undo the rubocop changes
